### DISCUSSION TITLE
How to include special directories? (`data`, `inst`, `tools`, `exec`, `po`)

### DISCUSSION THREAD

**Discussion by bicarlsen** [extendr/extendr#525] [Q&A]
What is the best way to include special ["Other components"](https://r-pkgs.org/misc.html) directories with a package? I tried placing the `inst` folder in my `R` directory, but it doesn't seem to be accessible from the package using [`system.file`](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/system.file).

`system.file(package = "my_package")` returns the path to the directory of the entire folder, but `system.file("my_file", package="my_package")` does not return the path to `my_file`. I have placed `my_file` in the root directory of the project `my_project/`, `my_project/inst/`, and `my_project/R/inst/`.

**Comment by CGMossa** [comment]
I don't know how to answer this.
Please put up a repository of this, or a similar setup, and I may look at
it.

I also don't know what this has to do with extendr. But it is fine if
you need help with this.
Note that the `inst` folder needs to reside in the root of your **R**
package.

**Reply by bicarlsen** [reply]
Here is an MRE repo: [`extendr-include-mre`](https://github.com/bicarlsen/extendr-includes-mre).

**Reply by multimeric** [reply]
Thanks for the MRE. I guess the obvious question to ask is, have you tried writing the exact same code in pure R and seeing what happens? That would factor out other possible package installation issues. I presume you have actually installed the package when you run this, because otherwise it won't work.

**Comment by CGMossa** [comment]
This project doesn't have an `inst`. I don't see what's failing to be
picked up by an R script where this package is installed.

**Reply by bicarlsen** [reply]
The `inst` folder was placed under the `R` folder. When placed in the project root everything worked.

**Comment by bicarlsen** [comment]
Okay, got it sorted. Perhaps I messed something else up before, but the with the `inst` folder at the root level of the package, it is working now.
